### PR TITLE
feat: Adds asynch capabilities 

### PR DESCRIPTION
**Pull request by preetipidatala**
Adds asynch capabilities to the transform function within the TextAnalyticsSDK class.

### CODE REVIEW

File: src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala [hunk 1]
```diff
@@ -35,13 +38,19 @@ abstract class TextAnalyticsSDKBase[T](val textAnalyticsOptions: Option[TextAnal
       .credential(new AzureKeyCredential(getSubscriptionKey))
       .endpoint(getEndpoint)
       .buildClient()
+  var timeout = Duration(10,"sec")
```
Comment (mhamilton723) [inline src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala]: You can use the existing parameters in one of the cog services base classes for doing this

File: src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala [hunk 2]
```diff
@@ -35,13 +38,19 @@ abstract class TextAnalyticsSDKBase[T](val textAnalyticsOptions: Option[TextAnal
...
 
   protected def transformTextRows(toRow: TAResponseV4[T] => Row)
                                  (rows: Iterator[Row]): Iterator[Row] = {
-    rows.map { row =>
-      val results = invokeTextAnalytics(getValue(row, text), getValue(row,lang))
-      Row.fromSeq(row.toSeq ++ Seq(toRow(results))) // Adding a new column
-    }}
+    val futures = rows.map { row =>
+      Future {
+        lazy val results = invokeTextAnalytics(getValue(row, text), getValue(row,lang))
```
Comment (mhamilton723) [inline src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala]: dont think you need lazy here as the future is call by name which is kindof nifty ;)

**Review by mhamilton723 — CHANGES_REQUESTED** [review]

File: src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala [hunk 3]
```diff
@@ -1,243 +1,250 @@
...
+
+  protected def transformTextRows(toRow: TAResponseV4[T] => Row)
+                                 (rows: Iterator[Row]): Iterator[Row] = {
+    val futures = rows.map { row =>
+      Future {
+        val results = invokeTextAnalytics(getValue(row, text), getValue(row,lang))
+        Row.fromSeq(row.toSeq ++ Seq(toRow(results))) // Adding a new column
+      }(ExecutionContext.global)
+    }
+    bufferedAwait(futures, getConcurrency, Duration(getTimeout,SECONDS))(ExecutionContext.global)
```
Comment (HanLiMS) [inline src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala:50]: could you add tests as discussed?

File: src/test/scala/com/microsoft/ml/spark/cognitive/split1/TextAnalyticsSDKSuite.scala [hunk 4]
```diff
@@ -50,6 +51,32 @@ class DetectedLanguageSuitev4 extends TestBase with DataFrameEquality with TextK
     assert(iso(0).get(0).toString == "en" && iso(0).get(1).toString == "es" &&
       iso(1).get(0).toString == "fr" && iso(1).get(1).toString == "zh")
   }
+
+  test("Asynch Concurrency Functionality") {
```
Comment (HanLiMS) [inline src/test/scala/com/microsoft/ml/spark/cognitive/split1/TextAnalyticsSDKSuite.scala:55]: name as: "async incorrect concurrency check " or something to have the information of the purpose of this test?

File: src/test/scala/com/microsoft/ml/spark/cognitive/split1/TextAnalyticsSDKSuite.scala [hunk 5]
```diff
@@ -50,6 +51,32 @@ class DetectedLanguageSuitev4 extends TestBase with DataFrameEquality with TextK
...
+        detector
+          .setConcurrency(badConcurrency)
+          .transform(df)
+          .select("output.result.name","output.result.iso6391Name")
+          .collect()
+      }
+    assert(caught.getMessage.contains("java.lang.IllegalArgumentException"))
+  }
+
+  test("Asynch Timeout Functionality") {
```
Comment (HanLiMS) [inline src/test/scala/com/microsoft/ml/spark/cognitive/split1/TextAnalyticsSDKSuite.scala:68]: can we add a test case for valid input as well?

File: src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala [hunk 6]
```diff
@@ -1,243 +1,250 @@
...
-
-    TAResponseV4[SentimentScoredDocumentV4](
-      sentimentResult,
-      error,
-      stats,
-      Some(resultCollection.getModelVersion))
-  }
-  override def outputSchema: StructType = SentimentResponseV4.schema
-}
+package com.microsoft.ml.spark.cognitive
```
Comment (HanLiMS) [inline src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala:1]: can we make sure we have a clean diff here for the TextAanlyticsSDK.scala file?

**Comment by HanLiMS** [comment]
Remove this one -- New PR submitted.

**Pull request closed without merging**